sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveShim.scala

```diff
@@ -757,6 +757,7 @@ private[client] class Shim_v0_13 extends Shim_v0_12 {
           case attr: Attribute => Some(attr)
           case Cast(child @ IntegralType(), dt: IntegralType, _)
               if Cast.canUpCast(child.dataType.asInstanceOf[AtomicType], dt) => unapply(child)
+          case Cast(child @ StringType(), DateType, _) => unapply(child)
```

Comment:
What if an input str is a special value (e.g., now, today, ...)?

Reply:
These special values handle by:

https://github.com/apache/spark/blob/6d625ccd5b5a76a149e2070df31984610629a295/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/finishAnalysis.scala#L82-L87

https://github.com/apache/spark/blob/a5e13acd19871831a93a5bdcbc99a9eb9f1aba07/sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveShim.scala#L660-L661